src/Data/List/NonEmpty.purs

```diff
@@ -0,0 +1,84 @@
+module Data.List.NonEmpty where
+
+import Prelude
+
+import Control.Alt (class Alt)
+
+import Data.Foldable (class Foldable)
+import Data.Generic (class Generic)
+import Data.List ((:))
+import Data.List as L
+import Data.Newtype (class Newtype, unwrap)
+import Data.NonEmpty (NonEmpty, (:|))
+import Data.NonEmpty as NE
+import Data.Maybe (fromMaybe)
+import Data.Traversable (class Traversable)
+import Data.Tuple (Tuple(..))
+import Data.Unfoldable (class Unfoldable, unfoldr)
+
+newtype NonEmptyList a = NonEmptyList (NonEmpty L.List a)
+
+toUnfoldable :: forall f a. Unfoldable f => NonEmptyList a -> f a
+toUnfoldable (NonEmptyList nel) =
+  unfoldr (\xs -> (\rec -> Tuple rec.head rec.tail) <$> L.uncons xs) (NE.oneOf nel)
+
+singleton :: forall a. a -> NonEmptyList a
+singleton = NonEmptyList <<< NE.singleton
+
+head :: forall a. NonEmptyList a -> a
+head (NonEmptyList (x :| _)) = x
+
+last :: forall a. NonEmptyList a -> a
+last (NonEmptyList (x :| xs)) = fromMaybe x (L.last xs)
+
+tail :: forall a. NonEmptyList a -> L.List a
+tail (NonEmptyList (_ :| xs)) = xs
+
+init :: forall a. NonEmptyList a -> NonEmptyList a

```

Comment:
The result isn't non-empty in general though. This gives the equivalent of `init [x] = [x]`.